### PR TITLE
feat(ses): synth-time naming validation for dedicatedIpPoolName

### LINKED ISSUE #28451: aws-ses: The naming convention for the dedicated IP pool cannot be confirmed.
### Describe the feature

The dedicated pool has [specific characters that can be used in its name.](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ses-dedicatedippool.html#cfn-ses-dedicatedippool-poolname)

I propose a feature for CDK to verify that it adheres to this naming convention.

### Use Case

If a name containing forbidden characters is set for the dedicated IP pool, no error occurs at the stage of ‘cdk synth’, but it fails during deployment. Therefore, I believe it would be preferable to trigger an error at the synth stage.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.115.0

### Environment details (OS name and version, etc.)

irrelevant

### PR DESCRIPTION
**Pull request by badmintoncryer**
This Pull Request introduces a new validation feature to the DedicatedIpPoolProps. It ensures that the dedicatedIpPoolName adheres to the specified naming conventions, enhancing data integrity and preventing runtime errors due to invalid names.

Closes #28451

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by badmintoncryer** [comment]
Exemption Request

I believe that integration tests are not necessary for this PR.

File: packages/aws-cdk-lib/aws-ses/lib/dedicated-ip-pool.ts [hunk 1]
```diff
@@ -74,6 +78,10 @@ export class DedicatedIpPool extends Resource implements IDedicatedIpPool {
       physicalName: props.dedicatedIpPoolName,
     });
 
+    if (props.dedicatedIpPoolName && !/^[a-z0-9_-]{0,64}$/.test(props.dedicatedIpPoolName)) {
+      throw new Error('Invalid dedicatedIpPoolName. The name must only include lowercase letters, numbers, underscores, hyphens, and must not exceed 64 characters.');
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-ses/lib/dedicated-ip-pool.ts:82]:
```suggestion
      throw new Error(`Invalid dedicatedIpPoolName "${props.dedicatedIpPoolName}". The name must only include lowercase letters, numbers, underscores, hyphens, and must not exceed 64 characters.`);
```

File: packages/aws-cdk-lib/aws-ses/test/dedicated-ip-pool.test.ts [hunk 2]
```diff
@@ -26,4 +26,11 @@ test('dedicated IP pool with scailingMode', () => {
...
   });
-});
\ No newline at end of file
+});
+
+test('dedicated IP pool with invalid name', () => {
+  // THEN
+  expect(() => new DedicatedIpPool(stack, 'Pool', {
+    dedicatedIpPoolName: 'invalidName',
+  })).toThrow('Invalid dedicatedIpPoolName. The name must only include lowercase letters, numbers, underscores, hyphens, and must not exceed 64 characters.');
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-ses/test/dedicated-ip-pool.test.ts:35]:
```suggestion
  })).toThrow('Invalid dedicatedIpPoolName "invalidName". The name must only include lowercase letters, numbers, underscores, hyphens, and must not exceed 64 characters.');
```

**Review by kaizencc — APPROVED** [review]
Hopefully my changes to the error message don't break the build :). Thanks @badmintoncryer

**Pull request merged**